detection_rules/devtools.py

```diff
@@ -152,43 +155,47 @@ def build_integration_docs(ctx: click.Context, registry_version: str, pre: str,
-    new_version = Version(ver).bump(major, minor, patch)
-
-    kibana_version = f"^{new_version}.0" if not patch else f"^{new_version}"
-    registry_version = f"{new_version}.0-dev.0" if not patch else f"{new_version}-dev.0"
+    pkg_data = load_etc_dump('packages.yml')['package']
+    kibana_ver = Version(*pkg_data["name"].split("."))
+    pkg_ver = Version.parse(pkg_data["registry_data"]["version"])
+    pkg_kibana_ver = Version.parse(pkg_data["registry_data"]["conditions"]["kibana.version"].lstrip("^"))
+    if major_release:
+        pkg_data["name"] = str(kibana_ver.bump_major()).rstrip(".0")
+        pkg_data["registry_data"]["conditions"]["kibana.version"] = f"^{pkg_kibana_ver.bump_major()}"
+        pkg_data["registry_data"]["version"] = str(pkg_ver.bump_major().bump_prerelease("beta"))
+    if minor_release:
+        pkg_data["name"] = str(kibana_ver.bump_minor()).rstrip(".0")
+        pkg_data["registry_data"]["conditions"]["kibana.version"] = f"^{pkg_kibana_ver.bump_minor()}"
+        pkg_data["registry_data"]["version"] = str(pkg_ver.bump_minor().bump_prerelease("beta"))
+        pkg_data["registry_data"]["release"] = maturity
+    if patch_release:
+        latest_patch_release_ver = find_latest_integration_version("security_detection_engine",
+                                                                   maturity, pkg_data["name"])
+        if maturity == "ga":
+            pkg_data["registry_data"]["version"] = str(latest_patch_release_ver.bump_patch())
+            pkg_data["registry_data"]["release"] = maturity
+        else:
+            pkg_data["registry_data"]["version"] = str(latest_patch_release_ver.bump_prerelease("beta"))
+            pkg_data["registry_data"]["release"] = maturity
 
     # print the new versions
-    click.echo(f"New package version: {new_version}")
-    click.echo(f"New registry data version: {registry_version}")
-    click.echo(f"New Kibana version: {kibana_version}")
-
-    if package:
-        # update package version
-        package_data["name"] = str(new_version)
-
-    if kibana:
-        # update kibana version
-        package_data["registry_data"]["conditions"]["kibana.version"] = kibana_version
-
-    if registry:
-        # update registry version
-        package_data["registry_data"]["version"] = registry_version
-        # update packages.yml
+    if major_release or minor_release:
+        click.echo(f"New Kibana version: {pkg_data['name']}")
+        click.echo(f"New package Kibana version: {pkg_data['registry_data']['conditions']['kibana.version']}")
+    click.echo(f"New package version: {pkg_data['registry_data']['version']}")
 
-    if package or kibana or registry:
-        save_etc_dump({"package": package_data}, "packages.yml")
+    # we only save major and minor version bumps
+    # patch version bumps are OOB packages and thus we keep the base versioning
+    return pkg_data if patch_release else save_etc_dump({"package": pkg_data}, "packages.yml")
```

Comment:
save_etc_dump returns what in this case?

Reply:
It will not return anything. Changed to always save regardless of flag.